_includes/components/section-menu.njk

```diff
@@ -1,41 +1,44 @@
-<nav class="section-menu">
-    <div class="centering layout">
-        <input type="search" placeholder="Suchen..." class="search-input" id="search-input" />
-    </div>
-    <ul class="section-menu-list">
-        {{ template "menu" (dict "page" .FirstSection "currentPage" .) }}
-    </ul>
-</nav>
+{% set allPages = collections.all | eleventyNavigation("/" + eleventyNavigation.section) %}
 
-{{ define "menu" }}
-    {{ $currPage := .currentPage }}
-    {{ range .page.Pages }}
-        <li class="{{ if eq . $currPage }}active{{ end }}">
-            <a href="{{ .RelPermalink }}">
-                <i class="icon fal fa-{{ (default "scroll" (default .Parent.Params.icon .Params.icon)) }}"></i>
+{% macro menu(pages) %}
+    {% for page in pages %}
+        <li class="{% if page.section === eleventyNavigation.section and page.url === eleventyNavigation.url %}active{% endif %}">
+            <a href="{{ page.url }}">
+                <i class="icon fal fa-{{ page.icon or "scroll" }}"></i>
                 <span class="title">
-                    {{ .Title }}
+                    {{ page.title | markdownify | safe }}
                 </span>
             </a>
 
-            {{ $hiddenAttr := "" }}
+            {% if page.section !== eleventyNavigation.section %}
+                {% set hiddenAttribute = "hidden" %}
+            {% endif %}
 
-            {{ if not (or (.IsAncestor $currPage) (eq .Permalink $currPage.Permalink)) }}
-                {{ $hiddenAttr = "hidden" }}
-            {{ end }}
-
-            {{ if .IsNode }}
-                <ul {{ $hiddenAttr }}>
-                    {{ template "menu" (dict "page" . "currentPage" $currPage) }}
+            {% if (not page.anchors) or (page.anchors.length === 0) %}
```

Comment:
@MaKleSoft any idea why `page.anchors` doesn't even exist, here? I can't figure out why, and I'm hoping it's something simple/basic.

Reply:
`page.anchors` is no longer provided explicitly by parsed automatically from the page content. You can reuse the logic from the Padloc page for that: https://github.com/padloc/site-new/blob/master/_lib/toc.js

Reply:
Gotcha, but I'm inside a loop here, hence the `page`. It's coming from `eleventyNavigation` so maybe that's what's not providing the front matter data.